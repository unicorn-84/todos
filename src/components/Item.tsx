import React from 'react';
import type { ITask } from '../types';
import clsx from 'clsx';

interface IItemProps extends React.HTMLAttributes<HTMLDivElement> {
  task: ITask;
  toggleTask: (id: number) => void;
  removeTask: (id: number) => void;
}

const Item: React.FC<IItemProps> = ({
  task: { id, title, complete },
  removeTask,
  toggleTask,
  ...props
}) => {
  return (
    <div
      className={clsx(
        'flex items-center justify-between space-x-3 lg:space-x-4',
        {
          'opacity-30': complete,
        },
      )}
      {...props}
    >
      <label className="label cursor-pointer w-full justify-start space-x-3 lg:space-x-4 p-0">
        <input
          className="checkbox checkbox-primary checkbox-md lg:checkbox-lg"
          type="checkbox"
          checked={complete}
          onChange={() => {
            toggleTask(id);
          }}
        />
        <span
          className={clsx('text-3xl lg:text-4xl break-words leading-none', {
            'line-through': complete,
          })}
        >
          {title}
        </span>
      </label>
      <button
        onClick={() => {
          removeTask(id);
        }}
        className="btn btn-warning btn-xs lg:btn-sm btn-square"
      >
        <svg
          xmlns="http://www.w3.org/2000/svg"
          className="h-6 w-6"
          fill="none"
          viewBox="0 0 24 24"
          stroke="currentColor"
        >
          <path
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            d="M6 18L18 6M6 6l12 12"
          />
        </svg>
      </button>
    </div>
  );
};

export default Item;
