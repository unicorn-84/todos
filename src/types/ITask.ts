interface ITask {
  taskId: number;
  taskTitle: string;
  taskComplete: boolean;
}

export default ITask;
