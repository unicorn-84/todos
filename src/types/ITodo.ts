interface ITodo {
  todoId: number;
  task: string;
  complete: boolean;
}

export default ITodo;
