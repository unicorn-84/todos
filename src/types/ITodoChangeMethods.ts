interface ITodoChangeMethods {
  addTodo: () => void;
  removeTodo: (id: number) => void;
  toggleTodo: (id: number) => void;
}

export default ITodoChangeMethods;
