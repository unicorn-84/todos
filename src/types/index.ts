export type { default as ITask } from './ITask';
