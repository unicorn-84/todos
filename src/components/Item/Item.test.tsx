import React from 'react';
import { render, screen } from '@testing-library/react';
import { type ITask } from '../../types';
import Item from '.';

const task: ITask = {
  id: 1,
  title: 'Task1',
  complete: true,
};

describe('Item', () => {
  test('should render correctly', () => {
    render(
      <Item
        data-testid="item"
        task={task}
        toggleTask={vi.fn()}
        removeTask={vi.fn()}
      />,
    );

    expect(screen.getByTestId('item')).toMatchSnapshot();
  });

  test('should render the task', () => {
    render(<Item task={task} toggleTask={vi.fn()} removeTask={vi.fn()} />);

    expect(screen.getByRole('checkbox')).toBeInTheDocument();
    expect(screen.getByText('Task1')).toBeInTheDocument();
    expect(screen.getByRole('button')).toBeInTheDocument();
  });
});
