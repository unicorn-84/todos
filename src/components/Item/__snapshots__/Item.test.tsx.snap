// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Item > should render correctly 1`] = `
<div
  class="flex items-center justify-between space-x-3 lg:space-x-4 opacity-30"
  data-testid="item"
>
  <div
    class="label cursor-pointer w-full justify-start space-x-3 lg:space-x-4 p-0"
  >
    <input
      class="checkbox checkbox-primary checkbox-md lg:checkbox-lg"
      type="checkbox"
    />
    <span
      class="text-3xl lg:text-4xl break-words leading-none line-through"
    >
      Task1
    </span>
  </div>
  <button
    class="btn btn-warning btn-xs lg:btn-sm btn-square"
  >
    <svg
      class="h-6 w-6"
      fill="none"
      stroke="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M6 18L18 6M6 6l12 12"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
      />
    </svg>
  </button>
</div>
`;
